tests/cpptests/micro-benchmarks/benchmark_index_iterator.cpp

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+#include "benchmark/benchmark.h"
+#include "iterator_util.h"
+#include "redismock/util.h"
+
+#include <random>
+#include <vector>
+
+#include "src/iterators/iterator_api.h"
+#include "src/iterators/inverted_index_iterator.h"
+
+#include "deprecated_iterator_util.h"
+#include "src/index.h"
+
+class BM_IndexIterator_Base : public benchmark::Fixture {
+public:
+    static bool initialized;
+    static constexpr size_t n_ids = 100'000;
+
+    std::vector<t_docId> ids;
+    InvertedIndex *index;
```

Comment:
Also here I miss a little parametrization on block sizes, seekers, etc...?

Reply:
The block size is derived from the index flags, and also the seekers/encoders/decoders.
We test all the valid flag combinations.